### PR TITLE
Fix compiletest build on nightly

### PR DESCRIPTION
**Pull request by brendanzab**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -3,36 +3,59 @@ sudo: false
...
+  - echo -en 'travis_fold:start:script.test.compile_test\\r'
+  - travis-cargo --only nightly test -- --features "test nightly" -p gluon compile_test
+  - echo -en 'travis_fold:end:script.test.compile_test\\r'
+
+  - echo -en 'travis_fold:start:script.bench\\r'
+  - travis-cargo --only nightly bench
+  - echo -en 'travis_fold:end:script.bench\\r'
+
+  - echo -en 'travis_fold:start:script.build\\r'
+  - cargo build --release
```
Comment (Marwes) [inline .travis.yml]: `cargo bench` already does `cargo build --release` so move this above `bench` (or it makes not sense for nightly

File: tests/compiletest.rs [hunk 2]
```diff
@@ -21,12 +21,11 @@ fn run_mode(mode: &'static str) {
...
 
     let mut config = compiletest::default_config();
-    let cfg_mode = mode.parse().ok().expect("Invalid mode");
 
-    config.mode = cfg_mode;
+    config.mode = mode.parse().ok().expect("Invalid mode");
     config.src_base = PathBuf::from(format!("tests/{}", mode));
     let dir = out_dir.to_str().unwrap();
-    config.target_rustcflags = Some(format!("-L {} -L {}/deps", dir, dir));
+    config.target_rustcflags = Some(format!("-L {}/build -L {}/deps", dir, dir));
```
Comment (Marwes) [inline tests/compiletest.rs]: what ends up in build?

Comment (brendanzab) [inline tests/compiletest.rs]: The gluon libraries

Comment (Marwes) [inline tests/compiletest.rs]: Huh, hadn't seen that dir before but it looks like it just contains the output of the build script (or have it changed in a recent nightly?).

Comment (brendanzab) [inline tests/compiletest.rs]: Hmm - doesn't seem to work actually :/

File: .travis.yml [hunk 3]
```diff
@@ -1,38 +1,20 @@
...
+    - secure: K26ZRkSrB2i9bUXgpIoMnKpgC1kItXDyvkIi1iRQE4aZxDOi7K43vO46heDcRrqVj+G5TLF3CTCS4eU7NW13tF2uh3a+r1Z8X1ZJuoVEnIAkMPQMm4GCnsRZUB23rBiNY1MB7CFCA8VhbLRCCgvr+OJZshb58XNZh2QSRGUY6MKbAAn11gCWvV7fl7CU0S65IveOxGQZLG8Rycb593hW8Vl1OEjYe8PD9yUz6kotIsiWqOcIWLF1bljoJFbTZMq07gp4n80wcx7OpeCFxHe9z0SOR81m3Jic2rHrvFNt72+lG7Exz6MavwHPENtotZ/81rvcg9F5dM0UJ6vaqnSbemXxQ8FGkuZImkIAbkyaKf8xwMx2++6mg76Sn8Y/dB4wc3PZlBE19834Zrtocp0t+vAmFmx/LDhJl5+sj3/vm+wbp317DVuQQCNeBqQoNarYxl/OpgNsbMhIYRDcK2SIPRp37E4PgYbW1/ilMNg46qis3O9tairjO6yJ7oep9OjNcDs8IQwUc9G47N63Xwpm6vhWPeCdC4LnRD+0eXce4C37W4q4CTSM5oYi7+hxV0vEGLWNV/YZdmaJpFoysJ774eIKevyOBRsSLqCMClU/rCVlJsFOzBLjYOaI6YARGDpHjcL2LbfNIBDSwe0yg5WkoIAA0EibPhDU9vedPSYFt5A=
```
Comment (brendanzab) [inline .travis.yml]: Do we still need this?

Comment (Marwes) [inline .travis.yml]: I believe this was just for doc uploads so no.

**Comment by Marwes** [comment]
Can't seem to reproduce the compiletest failure locally, did you manage to do it or is it just travis?

**Comment by brendanzab** [comment]
Just Travis I think

**Comment by Marwes** [comment]
Feeling a bit better today and managed to reproduce the build problem locally. Its not the most solid fix but it should work for all hours except 2 over the course of a year :). I believe that https://github.com/rust-lang/cargo/pull/3319 might provide the necessary functionality to do this in a better way later on.

**Comment by Marwes** [comment]
Aaand it still fails, seems like a made a mistake when testing locally...

**Comment by Marwes** [comment]
Ok, that should do it. `--extern lib_name=lib_path` was needed instead of `-l lib_path`

**Comment by brendanzab** [comment]
Decided just to get rid of the build system stuff for now. Will merge on success.

**Comment by Marwes** [comment]
The build system seemed fine to me but okay.

**Pull request merged**